x-pack/plugins/infra/public/components/asset_details/template/page.tsx

```diff
@@ -24,16 +26,46 @@ import { getIntegrationsAvailable } from '../utils';
 export const Page = ({ tabs = [], links = [] }: ContentTemplateProps) => {
   const { loading } = useAssetDetailsRenderPropsContext();
   const { metadata, loading: metadataLoading } = useMetadataStateContext();
-  const { rightSideItems, tabEntries, breadcrumbs } = usePageHeader(tabs, links);
+  const { rightSideItems, tabEntries, breadcrumbs: headerBreadcrumbs } = usePageHeader(tabs, links);
   const { asset } = useAssetDetailsRenderPropsContext();
   const { actionMenuHeight } = useKibanaHeader();
   const trackOnlyOnce = React.useRef(false);
 
   const { activeTabId } = useTabSwitcherContext();
   const {
-    services: { telemetry },
+    services: {
+      telemetry,
+      serverless,
+      observabilityShared: {
+        navigation: { PageTemplate },
+      },
+    },
   } = useKibanaContextForPlugin();
 
+  const parentBreadcrumbResolver = useParentBreadcrumbResolver();
+  const breadcrumbOptions = parentBreadcrumbResolver.getBreadcrumbOptions(asset.type);
+  const breadcrumbs = useMemo(
+    () => [
+      {
+        ...breadcrumbOptions.link,
+        text: breadcrumbOptions.text,
+      },
+      {
+        text: asset.name,
+      },
+    ],
+    [breadcrumbOptions, asset.name]
+  );
+
+  useMetricsBreadcrumbs(breadcrumbs);
+  useEffect(() => {
+    if (serverless) {
```

Comment:
Should we also use the `isServerlessEnv` from `useKibanaEnvironmentContext` here too?

Reply:
The result is the same, tbh. Since we need to use `serverless` plugin, I thought it would be redundant to check the `isServerlessEnv` flag.